go/arrow/csv/csv_test.go

```diff
@@ -161,35 +157,25 @@ func TestCSVReader(t *testing.T) {
 		},
 		nil,
 	)
-	r := csv.NewReader(bytes.NewReader(raw), schema,
-		csv.WithAllocator(mem),
-		csv.WithComment('#'), csv.WithComma(';'),
-	)
-	defer r.Release()
-
-	r.Retain()
-	r.Release()
-
-	if got, want := r.Schema(), schema; !got.Equal(want) {
-		t.Fatalf("invalid schema: got=%v, want=%v", got, want)
-	}
-
-	out := new(bytes.Buffer)
-
-	n := 0
-	for r.Next() {
-		rec := r.Record()
-		for i, col := range rec.Columns() {
-			fmt.Fprintf(out, "rec[%d][%q]: %v\n", i, rec.ColumnName(i), col)
-		}
-		n++
-	}
-
-	if got, want := n, 2; got != want {
-		t.Fatalf("invalid number of rows: got=%d, want=%d", got, want)
-	}
-
-	want := `rec[0]["bool"]: [true]
+	for _, tc := range []struct {
+		name    string
+		csv     io.Reader
+		opts    []csv.Option
+		records int
+		want    string
+	}{
+		{
+			name: "including various values which doesn't contain null values",
+			csv: bytes.NewBufferString(`## a simple set of data which contains all supported types
```

Comment:
I think this is better than using csv files in [csv/testdata](https://github.com/apache/arrow/tree/8cc92f4ab443a6bafe69dec0b0161bfcf349fb79/go/arrow/csv/testdata) directory for maintainability reasons.

If you agree, I will remove `csv/testdata` directory and create input csv buffer at each scenarios.
If you don't agree, I'll add `csv/testdata/types_with_null.csv` and use it.

How do you feel about this?